### DISCUSSION TITLE
Parsing `_embedded` items from HAL with Quarkus REST client

### DISCUSSION THREAD

**Discussion by kinkkong** [resteasy/resteasy#4248] [Q&A] [answered]
For Quarkus REST extension I can use `quarkus-rest-links` extension to support HAL format. There are two handy annotation to inject links @RestLink(rel = "list"), @InjectRestLinks. 

I'm trying to parse HAL body with REST Client.
```
{
    "_embedded": {
        "entity": [
....
}
```

Is there is any extension or mechanism to support that? I know that I can wrap it in response object line `response._embedded.entity1` but I would like some more generic mechanism.

**Comment by jamezp** [comment, marked as answer]
This looks like more of a question for [Quarkus](https://github.com/quarkusio/quarkus/discussions). I don't think `quarkus-rest-links` uses RESTEasy, but I could be wrong. Either way, RESTEasy itself will support JSON, but there is no built in support for HAL. TBH I'm not sure what HAL is in this context :)
